bolt-design-system/bolt
# Bolt Component Generator!

![cleanshot 2019-03-02 at 10 07 49](https://user-images.githubusercontent.com/1617209/53683848-e6d51100-3cd3-11e9-8b24-2eace5723183.gif)

## Jira
TBD

## Summary
Yeoman component generator for easily and consistently spinning up new Bolt components with the same base structure!

## Details
- Scaffolds out the new component's overall structure + basic files most components require
- Auto-adds / updates the `package.json` and `.boltrc` configs in the Bolt monorepo
- Also auto-adds the basic component docs page that shows up in Pattern Lab

Lots more to be done on this but super excited with how much we'll get out of this out of the gate!

## How to test
- Test running the new `yarn create-bolt-component` NPM script in the root of the Bolt repo and follow the steps to generate a new Bolt component (some example names to try out are `modal`, `carousel`, `dropdown-menu`, etc -- especially components with more than one word in the name!)
- A couple other questions things to think about with this initial chunk of work:
  - Are we missing anything?
  - Too much? Too little? Just right?
  - Are oddly worded component names that don't work as expected?
  - Are these new components generated usable enough (ie. they can show something on the page + confirm theres enough Twig, Scss, and Web Component structure to confirm a new generated component actually works)
  - Do we / should we auto-format the newly generated component (ie. auto lint the code after generating)?
